### PR TITLE
Add DynStringSet.

### PR DESCRIPTION
**Pull request by devnev**
DynStringSet provides easier lookup of individual values than DynStringSlice.

### CODE REVIEW

**Comment by devnev** [comment]
According to http://peter.bourgon.org/go-in-production/#formatting-and-style, a set should be implemented as `map[string]struct{}` instead of the `map[string]bool` I used in this PR. Hold off and I'll change it around.

File: dynstringset.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"unsafe"
+
+	flag "github.com/spf13/pflag"
+)
+
+// DynStringSet creates a `Flag` that represents `map[string]struct{}` which is safe to change dynamically at runtime.
+// Unlike `pflag.StringSlice`, consecutive sets don't append to the slice, but override it.
```
Comment (mwitkow) [inline dynstringset.go]: Um. The current `DynStringSlice` doesn't append, it overwrites as well.

File: dynstringset.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			return err
+		}
+	}
+	oldPtr := atomic.SwapPointer(&d.ptr, unsafe.Pointer(&s))
+	if d.notifier != nil {
+		go d.notifier(*(*map[string]struct{})(oldPtr), s)
+	}
+	return nil
+}
+
```
Comment (mwitkow) [inline dynstringset.go]: add an `Contains` method as a convenience.

**Comment by mwitkow** [comment]
ROFL

File: dynstringset.go (hunk 1 above)
Comment (devnev) [inline dynstringset.go]: Yes, and `DynStringSlice` also has an explicit comment to document that feature, so I presume it is worth documenting on `DynStringSet` too. Note that its being compared to `pflag.StringSlice`

File: dynstringset.go (hunk 2 above)
Comment (devnev) [inline dynstringset.go]: Done.

**Comment by devnev** [comment]
PTAL.

**Pull request merged**